statefun-sdk-js/src/egress.ts

```diff
@@ -0,0 +1,131 @@
+    }
+    if (typeof value === 'string') {
+        return Buffer.from(value);
+    }
+    if (Number.isSafeInteger(value)) {
+        let buf = Buffer.alloc(4);
+        buf.writeInt32BE(value);
+        return buf;
+    }
+    if (value instanceof Uint8Array) {
+        // this includes node's Buffer.
+        return Buffer.from(value);
+    }
+    throw new Error("Unable to deduce a type automatically. Please provide an explicit type, or string/Buffer as a value.");
+}
+
+export interface KafkaEgressOpts {
+    typename: string,
+    topic: string,
+    key?: string,
+    value: any,
+    valueType?: Type<any>
+}
+
+/**
+ * Build a message that can be emitted to a Kafka generic egress.
+ *
+ * @param typename
+ * @param topic
+ * @param key
+ * @param value
+ * @param valueType
+ * @returns {EgressMessage}
+ */
+function kafkaEgressMessage({typename = "", topic = "", key = "", value = null, valueType}: KafkaEgressOpts) {
+    if (isEmptyOrNull(typename)) {
+        throw new Error("typename is missing");
+    }
+    validateTypeName(typename);
+    if (isEmptyOrNull(topic)) {
+        throw new Error("topic is missing")
+    }
+    if (value === undefined || value === null) {
+        throw new Error("value is missing");
+    }
+    let pbKafka = new PB_KAFKA()
+    pbKafka.setTopic(topic);
+    pbKafka.setValueBytes(serialize(valueType, value));
+    if (!isEmptyOrNull(key)) {
+        pbKafka.setKey(key);
+    }
+    const bytes = pbKafka.serializeBinary();
+    const box = TypedValueSupport.toTypedValueRaw("type.googleapis.com/io.statefun.sdk.egress.KafkaProducerRecord", bytes);
+    return new EgressMessage(typename, box);
+}
+
+export interface KinesisEgressOpts {
+    typename: string;
+    stream: string,
+    partitionKey: string,
+    hashKey?: string;
+    value: any;
+    valueType?: Type<any>
+}
+
+function kinesisEgressMessage({typename = "", stream = "", partitionKey = "", hashKey = "", value = null, valueType}: KinesisEgressOpts) {
+    if (isEmptyOrNull(typename)) {
+        throw new Error("typename is missing");
+    }
+    validateTypeName(typename);
+    if (isEmptyOrNull(stream)) {
+        throw new Error("stream is missing");
+    }
+    if (isEmptyOrNull(partitionKey)) {
+        throw new Error("partition key is missing");
+    }
+    if (value === undefined || value === null) {
+        throw new Error("value is missing");
+    }
+    let record = new PB_KINESIS();
+    record.setStream(stream);
+    record.setPartitionKey(partitionKey);
+    record.setValueBytes(serialize(valueType, value))
+    if (!isEmptyOrNull(hashKey)) {
+        record.setExplicitHashKey(hashKey);
+    }
+    const bytes = record.serializeBinary();
+    const box = TypedValueSupport.toTypedValueRaw("type.googleapis.com/io.statefun.sdk.egress.KinesisEgressRecord", bytes);
+    return new EgressMessage(typename, box);
+}
+
+export {kafkaEgressMessage}
+export {kinesisEgressMessage}
+export {serialize as trySerializerForEgress}
```

Comment:
Personally I would try to avoid renaming the export, it makes things more confusing.